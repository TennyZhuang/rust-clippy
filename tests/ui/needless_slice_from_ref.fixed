// run-rustfix

#![warn(clippy::needless_slice_from_ref)]

fn main() {
    let x = 3;
    let _s = &[x];
    let _s = &[x];
}
