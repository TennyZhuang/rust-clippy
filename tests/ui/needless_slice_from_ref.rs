// run-rustfix

#![warn(clippy::needless_slice_from_ref)]

fn main() {
    let x = 3;
    let _s = core::slice::from_ref(&x);
    let _s = std::slice::from_ref(&x);
}
