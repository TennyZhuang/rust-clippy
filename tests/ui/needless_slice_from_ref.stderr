error: needless slice::from_ref
  --> $DIR/needless_slice_from_ref.rs:7:14
   |
LL |     let _s = core::slice::from_ref(&x);
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `&[x]`
   |
   = note: `-D clippy::needless-slice-from-ref` implied by `-D warnings`

error: needless slice::from_ref
  --> $DIR/needless_slice_from_ref.rs:8:14
   |
LL |     let _s = std::slice::from_ref(&x);
   |              ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `&[x]`

error: aborting due to 2 previous errors

